microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,156 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
+    max_pvalue: float
+        Maximal P-value of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on P-value.
+    max_evalue: float
+        Maximal E-value of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on E-value.
+    min_fragment_length: int
+        Minimal fragment length of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on fragment length.
+
+    Returns
+    -------
+    [(str, str)] where first component is a fasta header, while the second is
+    its fasta sequence.
+
+    Raises
+    ------
+    IOError
+        If the file cannot be written.
+
+    Notes
+    -----
+    A hit must satisfy ALL filtering options (min_prob, max_pvalue, max_evalue,
+    min_fragment_length) to be included in the resulting list.
+    """
+
+    # parse hits from file
+    hits = parse_pdb_match(hhsuite_fp)
+
+    # filter hits
+    if min_prob is not None:
+        hits = [hit for hit in hits if hit['Probab'] >= min_prob]
+    if max_pvalue is not None:
+        hits = [hit for hit in hits if hit['P-value'] <= max_pvalue]
+    if max_evalue is not None:
+        hits = [hit for hit in hits if hit['E-value'] <= max_evalue]
+    if min_fragment_length is not None:
+        hits = [hit for hit in hits if frag_size(hit) >= min_fragment_length]
+
+    # read the original protein file, used to run HHsearch
+    p = Protein.read(fullsequence_fp, seq_num=1)
+    queryname = p.metadata['id'] + " " + p.metadata['description']
+
+    results = {'match': [], 'non_match': []}
+    # select non overlapping positive hits
+    subseqs_pos = select_hits(hits, e_value_threshold=999999)
+
+    for hit in subseqs_pos:
+        _id = get_q_id(hit)
+        header = "%s_%i-%i" % (queryname,
+                               hit['alignment'][_id]['start'],
+                               hit['alignment'][_id]['end'])
+        seq = hit['alignment'][_id]['sequence'].replace('-', '')
+        results['match'].append((header, seq, hit['alignment'][_id]['start']))
+
+    # collect gaps between positive hits
+    subseqs_neg = report_uncovered_subsequences(subseqs_pos, str(p),
+                                                min_subseq_len=0)
+    for hit in subseqs_neg:
+        header = "%s_%i-%i" % (queryname,
+                               hit['start'],
+                               hit['end'])
+        seq = hit['sequence']
+        results['non_match'].append((header, seq, hit['start']))
+
+    # write sub-sequences to a multiple fasta file, sequences are un-wrapped
+    try:
+        # sort by start position
+        for type_ in results:
+            results[type_] = sorted(results[type_], key=lambda x: x[2])
+
+        if subsequences_fp is not None:
+            for type_ in results:
+                f = open('%s.%s' % (subsequences_fp, type_), 'w')
+                for res in results[type_]:
+                    f.write(">%s\n%s\n" % res[:2])
```

Comment:
I am confused here. There are two `%s`s and only one parameter `res[:2]`. Is `res[:2]` a scalar or list?